packages/ui/src/types/authenticator/stateMachine/event.ts

```diff
@@ -0,0 +1,26 @@
+export type InvokeActorEventTypes =
+  | 'done.invoke.signInActor'
+  | 'done.invoke.signUpActor'
+  | 'done.invoke.signOutActor'
+  | 'done.invoke.resetPasswordActor';
+
+export type AuthEventTypes =
+  | 'CHANGE'
+  | 'BLUR'
+  | 'FEDERATED_SIGN_IN'
+  | 'RESEND'
+  | 'RESET_PASSWORD'
+  | 'SIGN_IN'
+  | 'SIGN_OUT'
+  | 'SIGN_UP'
+  | 'SKIP'
+  | 'SUBMIT'
+  | 'INIT'
+  | InvokeActorEventTypes;
+
+export type AuthEventData = Record<PropertyKey, any>; // TODO: this should be typed further
```

Comment:
Any chance we can get do this TODO?

Reply:
https://github.com/aws-amplify/amplify-ui/pull/1386 ;)